frontend/src/hooks/query/use-vscode-url.ts

```diff
@@ -1,10 +1,16 @@
 import { useQuery } from "@tanstack/react-query";
 import OpenHands from "#/api/open-hands";
+import { useConversation } from "#/context/conversation-context";
 
 export const useVSCodeUrl = (config: { enabled: boolean }) => {
+  const { conversationId } = useConversation();
+
   const data = useQuery({
-    queryKey: ["vscode_url"],
-    queryFn: OpenHands.getVSCodeUrl,
+    queryKey: ["vscode_url", conversationId],
+    queryFn: () => {
+      if (!conversationId) throw new Error("No conversation ID");
```

Comment:
If you want to ensure that the query is not triggered unless `conversationId` is available, you can update the `enabled` property below (e.g., `enabled: !!conversationId && config.enabled`)

See [dependent queries](https://tanstack.com/query/latest/docs/framework/react/guides/dependent-queries) for more info